### DISCUSSION TITLE
Do dem.py or dloadOrbits.py requier a seperate licensed package?

### DISCUSSION THREAD

**Discussion by odito2** [isce-framework/isce2#518] [Q&A]
Hi, I'm new to ISCE and would like to know if it is possible to use dem.py for DEM downloading or dloadOrbits.py for orbits, without a seperate licensed package. I ask since I tried to download and failed. In case a seperate licensed package is needed, is there any other open source to download dem & orbits? How can I know which of the workflows of the open source ISCE needs a license? 
Thank you!

**Comment by riyasmjgeo** [comment]
Seperate license is not required. 
For orbit download of sentinel-1, you may find an alternative solution via https://anaconda.org/conda-forge/sentineleof

**Reply by odito2** [reply]
What of all the open ISCE source package is not open and requires license?

**Reply by EJFielding** [reply]
All of the ISCE2 package that is on GitHub is Open Source and does not require a license. There were some additional modules for ISCE2 written a while ago (insarApp.py and isceApp.py) that are not included and that is why the code prints a warning. Those old modules are now obsolete.

**Comment by odito2** [comment]
Thank you for your explanation. It is much clearer now.
I there any updated tutorial for ISCE? I found one from 2015, which is not
so relevant. I'm looking for some guidance.

Odelya


בתאריך יום ב׳, 18 ביולי 2022, 0:45, מאת Eric Fielding ‏<
***@***.***>:

**Reply by riyasmjgeo** [reply]
https://www.unavco.org/events/?category_name=short-courses&past_events=true&search=
You can find video lectures of past short courses

**Comment by yuankailiu** [comment]
Have anyone tried downloading DEM recently? The EARTHDATA server seems to have some issues?

For example, I do 
```
 dem.py -a stitch -b 24 25 117 119 -s 1 -m xml -r -c
```

Then it keeps trying to retrieve and keep failing due to timeout.

```
curl -n  -L -c $HOME/.earthdatacookie -b $HOME/.earthdatacookie -k -f -O http://e4ftl01.cr.usgs.gov/MEASURES/SRTMGL3.003/2000.02.11/N4
1W125.SRTMGL3.hgt.zip
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   267  100   267    0     0   1089      0 --:--:-- --:--:-- --:--:--  1089
  0     0    0     0    0     0      0      0 --:--:--  0:05:00 --:--:--     0
curl: (28) Operation timed out after 300207 milliseconds with 0 out of 0 bytes received
2022-07-20 16:38:58,699 - isce.contrib.demUtils.DemStitcher - WARNING - There was a problem in retrieving the file  http://e4ftl01.cr.
usgs.gov/MEASURES/SRTMGL3.003/2000.02.11/N41W125.SRTMGL3.hgt.zip. Exception
```

Hope it is not just for me. Any idea about where we can report this and get it fixed?

**Reply by EJFielding** [reply]
I saw this recently, but then it started working again. Maybe the LP DAAC server has some capacity problems. Also check that you are not hitting the problem with Python 3.10 (#506 and #458 ).

**Reply by yuankailiu** [reply]
Thanks for sharing your experience. I am using python 3.8.0, and it does not complain about the `SystemError: PY_SSIZE_T_CLEAN macro must be defined for '#' formats` error as in #458 and #506.

Right now just an USGS link issue. Wait for them to fix it. Thanks!

**Comment by EJFielding** [comment]
I believe that we found and fixed the main problem with #560 that has been merged. The USGS now requires `https` so we had to update the ISCE2 code. In addition, there are still some capacity issues that cause some downloads to be slow. One tile that I downloaded took 15 minutes.
